crates/client/rpc/src/lib.rs

```diff
@@ -731,4 +731,128 @@ where
 
         Ok(DeclareTransactionResult { transaction_hash: transaction.hash.into(), class_hash: FieldElement::ZERO })
     }
+
+    /// Returns a transaction details from it's hash.
+    ///
+    /// It first verifies if the transaction was included
+    /// in an older block, verifying db mapping in the client database.
+    /// If not found, it also verifies if the transaction is pending
+    /// in the runtime.
+    ///
+    /// # Arguments
+    ///
+    /// * `transaction_hash` - Hex string of the transaction hash.
+    fn get_transaction_by_hash(&self, transaction_hash: FieldElement) -> RpcResult<Transaction> {
+        let block_hash_from_db = self
+            .backend
+            .mapping()
+            .block_hash_from_transaction_hash(H256::from(transaction_hash.to_bytes_be()))
+            .map_err(|e| {
+                error!("Failed to get transaction's substrate block hash from mapping_db: {e}");
+                StarknetRpcApiError::TxnHashNotFound
+            })?;
+
+        let substrate_block_hash = match block_hash_from_db {
+            Some(block_hash) => block_hash,
+            None => {
+                // Not found in the local mapping tx_hash <> block_hash, we
+                // then check into the transaction pool.
```

Comment:
don't check in the mempool, this is the job of `starknet_pendingTransaction`

Reply:
Makes sense.. sorry not having this conclusion before.
Just pushed. :+1: